nixos/modules/misc/nixpkgs.nix

```diff
@@ -151,7 +115,7 @@ in
         ''
           { allowBroken = true; allowUnfree = true; }
         '';
-      type = configType;
+      type = types.submodule ../../../pkgs/top-level/config.nix;
```

Comment:
I'm debating whether it should be this or `types.deferredModule`, ie pass a whole unevaluated (sub)module to Nixpkgs, and therefore also, make the Nixpkgs entrypoint accept a module for `config`.

The latter is somewhat more in the spirit of turning the "Nixpkgs invocation interface" entirely into a module, which is an admittedly unclear goal of https://github.com/NixOS/nixpkgs/pull/231940.